### PR TITLE
fix: Do not deploy ib-kubernetes and SR-IOV Device plugin in OCP

### PR DESCRIPTION
**Pull request by e0ne**

### CODE REVIEW

**Comment by rollandf** [comment]
@ noama-nv Please review

File: hack/templates/samples/mellanox.com_v1alpha1_nicclusterpolicy.template [hunk 1]
```diff
@@ -57,35 +57,11 @@ spec:
           }
         ]
       }
-  sriovDevicePlugin:
```
Comment (noama-nv) [inline hack/templates/samples/mellanox.com_v1alpha1_nicclusterpolicy.template:60]: @e0ne  can we leave it just be with comment?

Comment (e0ne) [reply hack/templates/samples/mellanox.com_v1alpha1_nicclusterpolicy.template:60]: unfortunately, we can't do it for OCP

**Comment by noama-nv** [comment]
LGTM

**Review by noama-nv — APPROVED** [review]

**Comment by rollandf** [comment]
@e0ne please backport

**Pull request merged**